.github/actions/install-openssl/action.yml

```diff
@@ -0,0 +1,32 @@
+name: Install OpenSSL
+description: 'Install OpenSSL >= 3 if not already present.'
+
+runs:
+  using: composite
+  steps:
+    # The ubuntu runner already has OpenSSL > 3 and CMake can find it.
+    - name: Install for Ubuntu.
```

Comment:
The OpenSSL version on ubuntu 22.04 isn't going to go backwards (unless entropy in the universe reverses), so I didn't do anything smart here.